test/metabase/http_client.clj

```diff
@@ -153,18 +169,23 @@
 (defn build-request-map
   "Build the request map we ultimately pass to [[clj-http.client]]. Add user credential headers, specify JSON encoding,
   and encode body as JSON."
-  [credentials http-body]
-  (merge
-   {:accept       :json
-    :headers      {@#'mw.session/metabase-session-header
-                   (when credentials
-                     (if (map? credentials)
-                       (authenticate credentials)
-                       credentials))}
-    :cookie-policy :standard
-    :content-type :json}
-   (when (seq http-body)
-     {:body (json/generate-string http-body)})))
+  [credentials http-body request-options]
+  (let [content-type (or (get-in request-options [:headers "content-type"]) "application/json")]
+    (merge-with #(if (map? %1)
```

Comment:
IMO `merge` combined with a series of `(when ...)` is easier to read because I don't need to understand this lambda function.

Reply:
Ehmm but it's not the same? I'd have just `merge-with merge`, but it breaks if two maps have the same non-map top-level key, like `(merge-with merge {:a 1} {:a 2})`. So this lambda is saying "merge-with merge if vals are maps, else just merge". Let me see if extract this will make it clearer.